Lib/fractions.py

```diff
@@ -315,6 +314,13 @@ def from_decimal(cls, dec):
                 (cls.__name__, dec, type(dec).__name__))
         return cls(*dec.as_integer_ratio())
 
+    @classmethod
+    def _from_pair(cls, num, den):
```

Comment:
Let's use `numerator` and `denominator` - there's no reason to introduce a naming inconsistency with the `__new__` method. It would also be good to have a docstring for the benefit of code readers - that docstring should explain that the purpose here is not to store an unnormalized fraction, but to bypass the normalization in situations where it's not needed. It might also be good to emphasize that it's an internal implementation detail that shouldn't be depended on by external users.